interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -224,6 +224,72 @@ namespace cling {
     setCallbacks(std::move(AutoLoadCB));
   }
 
+  //Overloading the Interpreter constructor by passing one more argument, the pointer
+  //to the "master" Interpreter.
+  Interpreter::Interpreter(Interpreter *originalInterpreter, int argc, const char* const *argv,
```

Comment:
Instead of duplicating another Interpreter c'tor, can we factor most of the code out? Can we maybe have the existing one call this one or vice versa? Check "constructor delegation", see e.g. https://msdn.microsoft.com/en-us/library/dn387583.aspx.